project/src/test/java/com/google/sps/EventTest.java

```diff
@@ -0,0 +1,80 @@
+    private static final String TIME = "3:00PM-5:00PM";
+    private static final String VOLUNTEER_OPPORTUNITY_NAME = "Volunteer";
+    private static final int NUM_SPOTS_LEFT = 5;
+    private static final String EVENT_ID = "0883de79-17d7-49a3-a866-dbd5135062a8";
+    private static final VolunteeringOpportunity OPPORTUNITY = new VolunteeringOpportunity.Builder(EVENT_ID, VOLUNTEER_OPPORTUNITY_NAME, NUM_SPOTS_LEFT).build();
+    private static final Set <VolunteeringOpportunity> OPPORTUNITIES = new HashSet<VolunteeringOpportunity> (Arrays.asList(OPPORTUNITY));
+    private static final User USER1 = new User.Builder("USER1", "USER1@test.com").build();
+    private static final User USER2 = new User.Builder("USER2", "USER2@test.com").build();
+    private static final User USER3 = new User.Builder("USER3", "USER3@test.com").build();
+    private static final Set <User> ATTENDEES = new HashSet<User> (Arrays.asList(USER1, USER2));
+    private static final Set <User> NEW_ATTENDEES = new HashSet<User> (Arrays.asList(USER1, USER2, USER3));
+
+    /* User creating event
+	 * Verify Builder class is created with correct required and optional fields */
+	@Test
+	public void testEventBuild() {
+        Event event = new Event.Builder(EVENT_NAME, DESCRIPTION, LABELS, LOCATION, DATE, TIME, HOST)
+            .setOpportunities(OPPORTUNITIES).setAttendees(ATTENDEES).build();
+
+        Assert.assertEquals(event.getName(), EVENT_NAME);
+        Assert.assertEquals(event.getDescription(), DESCRIPTION);
+        Assert.assertEquals(event.getLabels(), LABELS);
+        Assert.assertEquals(event.getLocation(), LOCATION);
+        Assert.assertEquals(event.getDate(), DATE);
+        Assert.assertEquals(event.getTime(), TIME);
+        Assert.assertEquals(event.getHost(), HOST);
+        Assert.assertEquals(event.getOpportunities(), OPPORTUNITIES);
+        Assert.assertEquals(event.getAttendees(), ATTENDEES);
+	}
+    
+    /* User editing event
+	 * Verify Event mergeFrom setter sets required and optional field */
+    @Test
+	public void setEventFields() {
+        Event event = new Event.Builder(EVENT_NAME, DESCRIPTION, LABELS, LOCATION, DATE, TIME, HOST).build();
+        Event.Builder changedEventBuilder = event.toBuilder().setAttendees(ATTENDEES).build().toBuilder();
+        changedEventBuilder.mergeFrom(event);
+        event = changedEventBuilder.build();
+        event = event.toBuilder().setName(NEW_EVENT_NAME).build();
+
+        Assert.assertEquals(event.getAttendees(), ATTENDEES);
+        Assert.assertEquals(event.getName(), NEW_EVENT_NAME);
+    }
+
+    /* User adding info to event
+	 * Verify adding attendees */
+    @Test
+	public void addAttendeeField() {
+        Event event = new Event.Builder(EVENT_NAME, DESCRIPTION, LABELS, LOCATION, DATE, TIME, HOST).setAttendees(ATTENDEES).build();
+        Event changedEvent = event.toBuilder().addAttendee(USER3).build();
+
+        Assert.assertEquals(changedEvent.getAttendees(), NEW_ATTENDEES);
```

Comment:
Readers cannot understand the test without reading the constant value, because readers don't know USER3 is in NEW_ATTENDEES, which is against best practice of unit test.

https://docs.google.com/document/d/1M9kKw-Qgwmi-fhIDab-HCsK6sp53qRJQaM7aM3s4rXs/edit#heading=h.cxwr9nos46ee

I would recommend you to not use ATTENDEES and NEW_ATTENDEES here. Change it to something like this:

        Event event = new Event.Builder(EVENT_NAME, DESCRIPTION, LABELS, LOCATION, DATE, TIME, HOST).setAttendees({USER1, USER2}).build();
        Event changedEvent = event.toBuilder().addAttendee(USER3).build();

        Assert.assertEquals(changedEvent.getAttendees(), {USER1, USER2, USER3});

Reply:
Makes sense, thanks